app/src/components/organisms/SurveyForm.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useState } from 'react';
+import Stack from '@mui/material/Stack';
+import SurveyClassQuestion from 'components/organisms/SurveyClassQuestion';
+import Button from '@mui/material/Button';
+import FormControlLabel from '@mui/material/FormControlLabel';
+import Checkbox from '@mui/material/Checkbox';
+import TextField from '@mui/material/TextField';
+import Divider from '@mui/material/Divider';
+import { classData, formVals, vals } from '../shared/interfaces/surveyForm.interfaces';
+import { overallDefaults } from '../shared/constants/surveyForm.constants';
+
+function SurveyForm(props: { formData: classData[] }) {
+  const [disable, setDisabled] = useState(true);
+  const [values, setValues] = useState(() => {
+    const currentValues: formVals = {
+      relief: false,
+      explanation: '',
+      courses: {}
+    };
+
+    currentValues.courses = props.formData.reduce((obj: any, field) => {
+      obj[field.CourseID.subject + ' ' + field.CourseID.code] = {
+        ...overallDefaults
+      };
+      return obj;
+    }, {});
+
+    return currentValues;
+  });
+
+  const onSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    console.log(values);
+    e.preventDefault();
+    //TODO: submit values somewhere
+  };
+
+  const fieldChanged = (courseName: string, type: string, value: string) => {
+    setValues((currentValues) => {
+      currentValues.courses[courseName][type as keyof vals] = value;
+      return currentValues;
+    });
+  };
+
+  const handleCheckbox = (event: React.ChangeEvent<HTMLInputElement>, checkbox: boolean) => {
+    setValues((currentValues) => {
+      if (checkbox) {
+        currentValues.relief = event.target.checked;
+        setDisabled(!disable);
+      } else {
+        currentValues.explanation = event.target.value;
+      }
+      return currentValues;
+    });
+  };
+
+  return (
+    <div>
+      <form onSubmit={onSubmit}>
+        <Stack style={{ height: '500px', overflowY: 'scroll' }} spacing={4}>

```

Comment:
I would avoid using a static height, when I minimize my browser this overflows. You could either add a bottom margin on this stack with enough space for a static height bottom stack (the relief stuff), or just set it to have height 75% to keep it simple.

Reply:
height is now using a calculated value that should be 50%. Overflow does not like it when a percentage height is used so calc() was needed.